### PR TITLE
Added support for group contents search

### PR DESCRIPTION
**Pull request by MikeTschudi**
#658

### CODE REVIEW

File: packages/arcgis-rest-portal/src/util/generic-search.ts [hunk 1]
```diff
@@ -35,13 +35,20 @@ export function genericSearch<T extends IItem | IGroup | IUser>(
...
+  let path;
   switch (searchType) {
     case "item":
       path = "/search";
       break;
     case "group":
       path = "/community/groups";
+      if (
+        typeof search !== "string" &&
+        !(search instanceof SearchQueryBuilder) &&
```
Comment (tomwayson) [inline packages/arcgis-rest-portal/src/util/generic-search.ts]: so is this something that should be added to `SearchQueryBuilder`?

Comment (patrickarlt) [reply packages/arcgis-rest-portal/src/util/generic-search.ts]: @MikeTschudi is there a particular reason that we can't search group content with a string or `SearchQueryBuilder`?

Comment (MikeTschudi) [reply packages/arcgis-rest-portal/src/util/generic-search.ts]: Useful, but perhaps as a new issue, an enhancement. In the meantime, this change addresses what Alison and we were seeking with #658.

Comment (patrickarlt) [reply packages/arcgis-rest-portal/src/util/generic-search.ts]: Couldn't we just add a new search type like `groupContent` and then add a new `searchGroupContent()` method to https://github.com/Esri/arcgis-rest-js/blob/master/packages/arcgis-rest-portal/src/groups/search.ts? 
Implementation would probably be something like:

```ts
export interface ISearchGroupContentOptions extends ISearchOptions {
  q: string | SearchQueryBuilder;
  groupId: string;
}


export function searchGroupContent(options: ISearchGroupContentOptions): Promise<ISearchResult<IItem>> {
  return genericSearch<IItem>(options, "groupContent");
}
```

Then you could use `SearchQueryBuilder` as follows:

```ts
const query = new SearchQueryBuilder()
  .match("Web Mapping Application")
  .in("type")
  .match("Demo App");

searchGroupItems({
  groupId: "ABC...",
  q: query
}).then((results) => {
  console.log(request);
});
```

**Comment by tomwayson** [comment]
I agree w/ @patrickarlt, would you be able to make those changes @MikeTschudi.

**Comment by MikeTschudi** [comment]
We can't search group content via `genericSearch()` with a `string` or `SearchQueryBuilder` query or `ISearchOptions` options directly because none has a group id, so these input types have to be screened out. But all can be handled via the new `searchGroupContent()` and `ISearchGroupContentOptions` as Patrick's example above shows.

**Comment by MikeTschudi** [comment]
@patrickarlt, @tomwayson, would you please re-review?
codecov appears to have [passed](https://codecov.io/gh/Esri/arcgis-rest-js/pull/675/commits) but gotten stuck again

File: packages/arcgis-rest-portal/src/util/generic-search.ts [hunk 2]
```diff
@@ -35,14 +43,34 @@ export function genericSearch<T extends IItem | IGroup | IUser>(
...
+        path = `/content/groups/${search.groupId}/search`;
+      } else {
+        return Promise.reject({
+          query: options.params.q,
+          total: 0,
+          start: 1,
+          num: 1,
+          nextStart: -1,
+          results: [] as IItem[]
+        } as ISearchResult<IItem>);
```
Comment (patrickarlt) [inline packages/arcgis-rest-portal/src/util/generic-search.ts:71]:
```suggestion
        return Promise.reject(new Error("you must pass a `groupId` option to `searchGroupContent`"));
```

File: packages/arcgis-rest-portal/src/util/generic-search.ts [hunk 3]
```diff
@@ -35,14 +43,34 @@ export function genericSearch<T extends IItem | IGroup | IUser>(
...
+      // Need to have groupId property to do group contents search,
+      // cso filter out all but ISearchGroupContentOptions
+      if (
+        typeof search !== "string" &&
+        !(search instanceof SearchQueryBuilder) &&
+        search.groupId
+      ) {
+        path = `/content/groups/${search.groupId}/search`;
+      } else {
+        return Promise.reject({
```
Comment (patrickarlt) [inline packages/arcgis-rest-portal/src/util/generic-search.ts:64]: @MikeTschudi should this reject with an error?

File: packages/arcgis-rest-portal/test/groups/search.test.ts [hunk 4]
```diff
@@ -51,6 +55,42 @@ describe("groups", () => {
...
+    it("should catch search for group contents without group id", done => {
+      genericSearch(
+        {
+          q: "water"
+        },
+        "groupContent"
+      ).then(
+        () => fail(),
+        err => {
+          expect(err).toEqual(EmptyGroupSearchResponse);
```
Comment (patrickarlt) [inline packages/arcgis-rest-portal/test/groups/search.test.ts:89]: @MikeTschudi since this throws an error now should this test change?

Comment (MikeTschudi) [reply packages/arcgis-rest-portal/test/groups/search.test.ts:89]: @patrickarlt, thank you

**Comment by tomwayson** [comment]
I will merge and release this Monday.

**Pull request merged**